docs/architecture/adr-010-remove-wire-msg-pay-for-blob.md

```diff
@@ -0,0 +1,149 @@
+
+With the introduction of [ADR 008: square size independent message commitments](./adr-008-square-size-independent-message-commitments.md), a user no longer needs to create, sign, and send multiple signatures. This enables us to reduce the complexity of the malleation process by removing `WireMsgPayForBlob` entirely. Instead, users will create and publish a `BlobTx` to the mempool. The `BlobTx` will include a `sdk.Tx` which will remain unmodified and end up on-chain.
+
+// TODO: diagram the transaction flow for a BlobTx from user -> block proposer -> tx in a block
+
+### `MalleatedTx`'s OriginalTxHash
+
+celestia-core contains a patch to replace the tx hash of a tx containing a `MsgPayForBlob` with the tx hash from the original `WireMsgPayForBlob` it was derived from. This change was needed because the transaction the user creates (one containing a single `WireMsgPayForBlob`) will always be different from the transaction that is included in a block (one containing a single `MsgPayForBlob`). Since the transactions are different, the hashes will also be different. This means Tendermint's default transaction indexing can't confirm to a user that their transaction was included in a block.
+
+See [celestia-core#607](https://github.com/celestiaorg/celestia-core/pull/607) and Tendermint docs on [indexing transactions](https://docs.tendermint.com/v0.34/app-dev/indexing-transactions.html).
+
+## Alternative Approaches
+
+Preserve existing `WireMsgPayForBlob` and `MsgPayForBlob`.
+
+## Decision
+
+Proposed
+
+## Detailed Design
+
+### Transaction Flow
+
+This transaction flow only describes the support for one tx with one message with one blob.
+
+Assume a user wants to publish the data "hello world" to the namespace: `11111111`.
+
+1. User creates a `MsgPayForBlob`
+
+    ```golang
+    type MsgPayForBlob struct {
+        Signer string
+        NamespaceID []byte // 11111111
+        BlobSize uint64 // len("hello world")
+        ShareCommitment []byte // e945c0bd85c106990...
+        ShareVersion uint32 // 0
+    }
+    ```
+
+2. The user takes this `MsgPayForBlob` and includes it as the sole message in a transaction (henceforth known as `MsgPayForBlobTx`).
+3. The user signs the `MsgPayForBlobTx`.
+4. The user marshals the `MsgPayForBlobTx` to bytes and includes it as a field in a new transaction. The new transaction is a `BlobTx` which includes an additional field for the data they wish to publish.
+
+    ```golang
+    type BlobTx struct {
+        Tx []byte // marshalled sdk.Tx that includes one MsgPayForBlob
+        Blob []byte // []byte("hello world")
+    }
+    ```
+
+5. The user signs the `BlobTx` and publishes it to a celestia-app consensus full node or validator and eventually lands in the mempool.
```

Comment:
Great question! I don't think the user needs to sign the BlobTx because (as you pointed out) if an attacker modifies the blob, the ShareCommitment won't match the modified blob.

@evan-forbes thoughts?

Reply:
apologies for missing this, yeah we don't have to sign the blobTx